universum/api.py

```diff
@@ -1,50 +1,54 @@
+from types import TracebackType
+from typing import Optional
+
 import sys
 
 from .modules.api_support import ApiSupport
 from .lib.gravity import Module, Dependency
+from .lib.module_arguments import ModuleArgumentParser
 from .lib.utils import format_traceback
 
 __all__ = ["Api"]
 
 
 class Api(Module):
-    description = "Universum API"
+    description: str = "Universum API"
     api_support_factory = Dependency(ApiSupport)
 
     @staticmethod
-    def define_arguments(parser):
+    def define_arguments(parser: ModuleArgumentParser) -> None:
         parser.add_argument('action', choices=["get-shelves", "file-diff", "swarm"],
                             help="Input some description")
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args, **kwargs) -> None:
         super().__init__(*args, **kwargs)
         try:
-            self.api_support = self.api_support_factory(api_mode=True)
-        except EnvironmentError as error:
-            sys.stderr.write(str(error) + '\n')
+            self.api_support: ApiSupport = self.api_support_factory(api_mode=True)
+        except EnvironmentError as e:
+            sys.stderr.write(str(e) + '\n')
             sys.exit(2)
 
         class MinimalOut:
             @staticmethod
-            def log(line):
+            def log(line: str) -> None:
                 pass
 
             @staticmethod
-            def report_build_problem(problem):
+            def report_build_problem(problem: str) -> None:
                 pass
 
             @staticmethod
-            def log_exception(error):
-                ex_traceback = sys.exc_info()[2]
-                sys.stderr.write("Unexpected error.\n" + format_traceback(error, ex_traceback))
+            def log_exception(exc: Exception) -> None:
```

Comment:
note: actual Output class uses 'str' here
maybe we should align the two

Reply:
Sure, it definitely should expect string.

Reply:
I'll change it in a separate PR